rust/src/csv/writer.rs

```diff
@@ -0,0 +1,139 @@
+                        write_primitive_array!(self.w, batch, row_index, col_index, Int32Array)
+                    }
+                    &DataType::Int64 => {
+                        write_primitive_array!(self.w, batch, row_index, col_index, Int64Array)
+                    }
+                    &DataType::UInt8 => {
+                        write_primitive_array!(self.w, batch, row_index, col_index, UInt8Array)
+                    }
+                    &DataType::UInt16 => {
+                        write_primitive_array!(self.w, batch, row_index, col_index, UInt16Array)
+                    }
+                    &DataType::UInt32 => {
+                        write_primitive_array!(self.w, batch, row_index, col_index, UInt32Array)
+                    }
+                    &DataType::UInt64 => {
+                        write_primitive_array!(self.w, batch, row_index, col_index, UInt64Array)
+                    }
+                    &DataType::Float32 => {
+                        write_primitive_array!(self.w, batch, row_index, col_index, Float32Array)
+                    }
+                    &DataType::Float64 => {
+                        write_primitive_array!(self.w, batch, row_index, col_index, Float64Array)
+                    }
+                    &DataType::Utf8 => {
+                        let array = batch
+                            .column(col_index)
+                            .as_any()
+                            .downcast_ref::<BinaryArray>()
+                            .unwrap();
+                        self.w.write("\"".as_bytes()).unwrap();
+                        self.w.write(array.get_value(row_index)).unwrap();
+                        self.w.write("\"".as_bytes()).unwrap();
+                    }
+                    other => panic!("unsupported type {:?}", other),
+                }
+            }
+            self.w.write("\n".as_bytes()).unwrap();
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use datatypes::{Field, Schema};
+    use std::sync::Arc;
+
+    #[test]
+    fn test_write_csv() {
+        let schema = Schema::new(vec![
+            Field::new("city", DataType::Utf8, false),
+            Field::new("lat", DataType::Float64, false),
+            Field::new("lng", DataType::Float64, false),
+        ]);
+
+        let cities = BinaryArray::from(vec![
+            "Elgin, Scotland, the UK",
+            "Stoke-on-Trent, Staffordshire, the UK",
+            "Solihull, Birmingham, UK",
+        ]);
+        let lat = PrimitiveArray::from(vec![57.653484, 53.002666, 52.412811]);
+        let lng = PrimitiveArray::from(vec![-3.335724, -2.179404, -1.778197]);
+
+        let batch = RecordBatch::new(
+            Arc::new(schema),
+            vec![Arc::new(cities), Arc::new(lat), Arc::new(lng)],
+        );
+
+        let file = File::create("/tmp/uk_cities.csv").unwrap();
+
+        let mut writer = Writer::new(file);
+        writer.write(&batch);
```

Comment:
Should we check whether result is what we expected and is in the correct format?